src/rustdoc_gen.rs

```diff
@@ -189,11 +210,30 @@ pub(crate) enum CrateType<'a> {
     },
 }
 
+#[derive(Debug, Clone)]
+pub(crate) struct FeatureConfig {
+    pub(crate) default_features: bool,
+    pub(crate) only_explicit_features: bool,
+    pub(crate) feature: Vec<String>,
+    pub(crate) all_features: bool,
+}
```

Comment:
Please document what these fields mean and provide a few examples that clarify how these values combine. For example, what does it mean to have `default_features = true` together with `feature` set to a non-empty list?

Also, since the clap interface says a few of these conflict with each other, might it be possible that an enum would be a better representation here?